### PR TITLE
fix: Stop search_generate from segfaulting

### LINKED ISSUE #3133: Search examples under library fail to run
Both test_search and search_generate segfault when running

### PR DESCRIPTION
**Pull request by bassmang**
Resolves #3133

### CODE REVIEW

File: library/search_generate.cc [hunk 1]
```diff
@@ -476,11 +476,12 @@ void train()
...
-    run_istream(gen, "phrase-table.tr", true);
-    run_istream(gen, "phrase-table.tr", false, 300000);
-    run_istream(gen, "phrase-table.te", false, 100000);
+    //run_istream(gen, "phrase-table.tr", true);
+    //run_istream(gen, "phrase-table.tr", false, 300000);
+    //run_istream(gen, "phrase-table.te", false, 100000);
+    run_easy();
   }
-  VW::finish(vw_obj);
+  VW::finish(vw_obj, false);
```
Comment (jackgerrits) [inline library/search_generate.cc:484]: What calls delete now?

Comment (bassmang) [reply library/search_generate.cc]: It wasn't being deleted. Just updated it to delete and commented out this line `Generator gen(vw_obj, nullptr); // &dict);` which was causing a memory freeing error

Comment (jackgerrits) [reply library/search_generate.cc]: But wont calling finish with false mean it wont be deleted?
```
void finish(VW::workspace& all, bool delete_all)
```

Comment (bassmang) [reply library/search_generate.cc]: Yes I removed `false`

**Review by jackgerrits — APPROVED** [review]

**Pull request merged**